decko-rails/lib/decko/rails.rb

```diff
@@ -1,11 +1,33 @@
 DECKO_RAILS_GEM_ROOT = File.expand_path('../../..', __FILE__)
 
+require 'rails/all'
+require 'decko/engine'
+
 module Decko
-  module Rails
+  module Rails # not sure we need this
     class << self
       def gem_root
         DECKO_RAILS_GEM_ROOT
       end
     end
   end
-end
\ No newline at end of file
+
+  if defined? ::Rails::Railtie
```

Comment:
Sure, if either of them get any more complex.  With this little logic, it hardly seems worth it.   

And yes about why this might be needed and when.